client/src/client.rs

```diff
@@ -885,6 +885,77 @@ pub trait RpcApi: Sized {
         )
     }
 
+    /// Attempts to add or remove a node from the addnode list.
+    /// Or try a connection to a node once.
+    /// Nodes added using addnode (or -connect) are protected from DoS disconnection and are not required to be full nodes/support SegWit as other outbound peers are (though such peers will not be synced from).
+    fn add_node(&self, addr: &str) -> Result<()> {
+        self.call("addnode", &[into_json(&addr)?, into_json("add")?])
+    }
+
+    fn remove_node(&self, addr: &str) -> Result<()> {
+        self.call("addnode", &[into_json(&addr)?, into_json("remove")?])
+    }
+
+    fn onetry_node(&self, addr: &str) -> Result<()> {
+        self.call("addnode", &[into_json(&addr)?, into_json("onetry")?])
+    }
+
+    /// Immediately disconnects from the specified peer node.
+    fn disconnect_node(&self, addr: &str) -> Result<()> {
+        self.call("disconnectnode", &[into_json(&addr)?])
+    }
+
+    fn disconnect_node_by_id(&self, node_id: u32) -> Result<()> {
+        self.call("disconnectnode", &[into_json("")?, into_json(node_id)?])
+    }
+
+    /// Returns information about the given added node, or all added nodes (note that onetry addnodes are not listed here)
+    fn get_added_node_info(&self, node: Option<&str>) -> Result<Vec<json::GetAddedNodeInfoResult>> {
+        if let Some(addr) = node {
+            self.call("getaddednodeinfo", &[into_json(&addr)?])
+        } else {
+            self.call("getaddednodeinfo", &[])
+        }
+    }
+
+    /// Return known addresses which can potentially be used to find new nodes in the network
+    fn get_node_addresses(
+        &self,
+        count: Option<usize>,
+    ) -> Result<Vec<json::GetNodeAddressesResult>> {
+        let cnt = count.unwrap_or(1);
+        self.call("getnodeaddresses", &[into_json(&cnt)?])
+    }
+
+    /// List all banned IPs/Subnets.
+    fn list_banned(&self) -> Result<Vec<json::ListBannedResult>> {
+        self.call("listbanned", &[])
+    }
+
+    /// Clear all banned IPs.
+    fn clear_banned(&self) -> Result<()> {
+        self.call("clearbanned", &[])
+    }
+
+    /// Attempts to add or remove an IP/Subnet from the banned list.
+    fn add_ban(&self, subnet: &str, bantime: Option<u64>, absolute: Option<bool>) -> Result<()> {
+        let bantime = bantime.unwrap_or(0);
```

Comment:
It seems odd to duplicate defaults here. Ideally we would use the key-value pairs that are left out if `None` was supplied. I think `bitcoind` supports that (at least `bitcoin-cli` does). What do you think @stevenroose or anyone more familiar with this repo?

Reply:
I had to look at https://developer.bitcoin.org/reference/rpc/setban.html to understand what you mean. I agree that it's odd to default to this here while Bitcoin Core defaults to these values too. 

I think it's cleaner to remove the `bantime` and `absolute` `Option`s here. Allows the user to explicitly specify what he wants without risking that this implementation or Bitcoin Core's RPC API changes.

If the `Option`s are not dropped, the defaults should be documented.

Reply:
Alright, I removed the `Option`s in https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/171/commits/8b92fa3f82a300f3291b87a0e079d8266dd94dd2.